source/extensions/filters/http/ext_proc/processor_state.cc

```diff
@@ -165,7 +186,7 @@ absl::Status ProcessorState::handleHeadersResponse(const HeadersResponse& respon
         // Fall through if there was never a body in the first place.
         ENVOY_LOG(debug, "The message had no body");
       } else if (complete_body_available_ && body_mode_ != ProcessingMode::NONE) {
-        if (callback_state_ != CallbackState::StreamedBodyCallback) {
+        if (callback_state_ == CallbackState::Idle) {
```

Comment:
previously this would be called for any state other than StreamedBodyCallback; now it only gets called for 'Idle'.

Is that becuase there used to be exactly 2 states and now there are 3?

Because if there were 3 or more states before, then this change seemes like it might have unintended consequences.